nx/lib/nx/lin_alg.ex

```diff
@@ -1148,9 +1148,13 @@ defmodule Nx.LinAlg do
 
     is_zero = diag |> Nx.equal(0) |> Nx.any?() |> Nx.to_number() |> Kernel.==(1)
 
-    iota = Nx.iota({n})
+    iota = Nx.iota({n}, type: type, backend: backend)
```

Comment:
If `determinant` was a defn, then this would not be necessary. Should we convert it to a `defn`?

Reply:
I'll leave this to a follow-up PR